src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -851,32 +859,30 @@ final void ensureWindowUpdated(DataFrame df) {
         }
     }
 
-    private record PushContinuationState(HeaderDecoder pushContDecoder, PushPromiseFrame pushContFrame) {}
-    private volatile PushContinuationState pcs;
-
     private <T> void handlePushPromise(Stream<T> parent, PushPromiseFrame pp)
         throws IOException
     {
         // always decode the headers as they may affect connection-level HPACK
         // decoding state
-        assert pcs == null;
+        assert pushContinuationState == null;
         HeaderDecoder decoder = new HeaderDecoder();
         decodeHeaders(pp, decoder);
         int promisedStreamid = pp.getPromisedStream();
         if (pp.endHeaders()) {
             completePushPromise(promisedStreamid, parent, decoder.headers());
         } else {
-            pcs = new PushContinuationState(decoder, pp);
+            pushContinuationState = new PushContinuationState(decoder, pp);
         }
     }
 
     private <T> void handlePushContinuation(Stream<T> parent, ContinuationFrame cf)
             throws IOException {
-        decodeHeaders(cf, pcs.pushContDecoder);
+        decodeHeaders(cf, pushContinuationState.pushContDecoder);
```

Comment:
I suggest declaring a local variable here to avoid reading pushContinuationState more than once.
Something like:

```
var pcs = pushContinuationState;
```
then use `pcs` wherever needed in that method.